src/Parser.php

```diff
@@ -67,7 +67,7 @@ public function getEnvironment()
 
     public function getVarName()
     {
-        return sprintf('__internal_%s', hash('sha256', __METHOD__.$this->stream->getSourceContext()->getCode().$this->varNameSalt++));
+        return sprintf('__internal_%s', hash(\PHP_VERSION_ID < 80100 ? 'sha256' : 'xxh128', __METHOD__.$this->stream->getSourceContext()->getCode().$this->varNameSalt++));
```

Comment:
The `varNameSalt` make it sure we don't generate the same string 2 times. Using a basic counter is enough for that.
```suggestion
        return sprintf('__internal_parser_%s', $this->varNameSalt++);
```

Reply:
Done in an other PR. https://github.com/twigphp/Twig/pull/3601